### DISCUSSION TITLE
RemoteFileServiceContribution example

### DISCUSSION THREAD

**Discussion by mememeyaw** [eclipse-theia/theia#13544] [Q&A] [answered]
Hello! I am really new to Theia and have a question regarding integrating remote file service. 

**_TL;DR: Does any know how to implement a RemoteFileServiceContribution and RemoteFileSystemProvider?_** 

I planned to integrate Theia to a backend server filesystem, meaning I want to have Theia access an API from a server for all file management purposes. Below is the planned architecture

![image](https://github.com/eclipse-theia/theia/assets/165215545/ee4bad4a-e68b-4f99-89a9-eafe498284bd)

Then I stumbled upon the [filesystem](https://github.com/eclipse-theia/theia/tree/master/packages/filesystem) extension, which might serve the purpose for this since it has [RemoteFileServiceContribution](https://eclipse-theia.github.io/theia/docs/next/classes/filesystem.RemoteFileServiceContribution.html). Unfortunately, I am a little bit unsure how to create a new contribution using RemoteFileServiceContribution. Does anyone have a short info or example for this? 

Thank you in advance!

**Comment by msujew** [comment, marked as answer]
Hey @mememeyaw,

You can implement the [corresponding interface](https://github.com/eclipse-theia/theia/blob/52ac2a4855874fbf4661f591c3c72f8b28e24e99/packages/filesystem/src/common/remote-file-system-provider.ts#L40) yourself and rebind the symbol on the frontend.

On another note: Do you use the vscode extension support? Since vscode extensions mainly live on the backend, they need the workspace files to be available on the machine that runs Theia to run correctly.

We also have a fairly new frontend-only mode, that might make sense for you. It is designed for use cases in which Theia mainly acts as a view on some source code (similar to vscode.dev). However, it currently does not offer vscode extension support.

**Reply by mememeyaw** [reply]
Hello @msujew! Thank you for the reply. Currently, I don't plan to support any vscode extension, but I might write Theia or headless plugin for LSP. I hope they do not need any locally available workspace files. 

I am interested with the new frontend-only mode. Do you have any documentation on this regard? I only knew it since it was mentioned in one of EclipseSource YouTube video but I didn't manage to find any documentation page.

**Reply by msujew** [reply]
We don't have any documentation on it yet. The best link I can provide is the contributing PR at https://github.com/eclipse-theia/theia/pull/12853.
